plugins/api/include/irods/plugins/api/switch_user_types.h

```diff
@@ -1,15 +1,82 @@
 #ifndef IRODS_SWITCH_USER_TYPES_H
 #define IRODS_SWITCH_USER_TYPES_H
 
+#include "irods/objInfo.h"
+
+/// The input data type used to invoke rc_switch_user.
+///
+/// \since 4.3.1
 typedef struct SwitchUserInput // NOLINT(modernize-use-using)
 {
-    // NOLINTBEGIN(modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+    // NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+    /// The part of a fully-qualified iRODS username preceding the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::userName) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the zone
+    ///
+    /// \since 4.3.1
     char username[64];
+
+    /// The part of a fully-qualified iRODS username following the pound sign.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
     char zone[64];
-    // NOLINTEND(modernize-avoid-c-arrays)
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// TODO
+    ///
+    /// \since 4.3.1
+    struct KeyValPair options;
 } switchUserInp_t;
 
 // NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
-#define SwitchUserInp_PI "str username[64]; str zone[64];"
+#define SwitchUserInp_PI "str username[64]; str zone[64]; struct KeyValPair_PI;"
+
+//
+// Options
+//
+
+/// TODO
+/// Influences how the server handles the proxy user.
+///
+/// If set to 1, the proxy user is switched too. This option will result in
+/// losing the ability to invoke rc_switch_user if this option is set to 1 and
+/// the user to switch to is not a \p rodsadmin. This option is not used when
+/// handling server-to-server connections.
+///
+/// \since 4.3.1
+#define KW_SWITCH_PROXY_USER            "switch_proxy_user"
+
+/// TODO
+///
+/// \since 4.3.1
+#define KW_CLOSE_OPEN_REPLICAS          "close_open_replicas"
+
+/// TODO
+/// Influences how the server handles server-to-server connections.
+///
+/// If set to 1, all connections created through redirection will be closed.
+/// If set to 0, rc_switch_user is invoked on all server-to-server connections.
+///
+/// If the server-to-server connection represents a connection to an iRODS
+/// server earlier than 4.3.1, the connection is closed.
+///
+/// \since 4.3.1
+#define KW_CLOSE_SVR_TO_SVR_CONNECTIONS "close_svr_to_svr_connections"
```

Comment:
How about we invert the meaning of this and rename it to one of the following ...
- `KW_KEEP_SVR_TO_SVR_CONNECTIONS`
- `KW_DO_NOT_CLOSE_SVR_TO_SVR_CONNECTIONS`

Are there other phrases?

Doing this allows the API plugin to react based on the existence of the keyword rather than its value.

Reply:
i like the idea of allowing 'just existence'... 

- `KW_KEEP_SVR_TO_SVR_CONNECTIONS`
- `KW_RETAIN_SVR_TO_SVR_CONNECTIONS`

hm.

Reply:
Either of those work and simplifies the API.

Thinking of going with `KEEP` only because it's shorter. Do you feel `RETAIN` is clearer in this case?

This keyword will cause the server to invoke `rc_switch_user` on all svr-to-svr connections if possible.

Reply:
`KEEP` is fine/great.

Reply:
done